File: src/main/java/ez/spring/vertx/ActiveProfiles.java
package ez.spring.vertx;

import java.util.Collection;
import java.util.HashSet;
import java.util.stream.Collectors;

/**
 * active profiles(ignore character cases)
 */
public class ActiveProfiles extends HashSet<String> {
    ActiveProfiles(Collection<String> values) {
        super(values.stream().map(String::toLowerCase).collect(Collectors.toSet()));
    }

    public boolean isDev() {
        return contains("dev") || contains("development");
    }

    public boolean isProd() {
        return contains("prod") || contains("production");
    }

    public boolean isTest() {
        return contains("test");
    }
}

File: src/main/java/ez/spring/vertx/ActiveProfiles.kt
package ez.spring.vertx

import org.springframework.context.ApplicationContext

/**
 * active profiles from [ApplicationContext.getEnvironment] (ignore character cases)
 */
class ActiveProfiles private constructor(private val innerSet: Set<String>) : Set<String> by innerSet {
    constructor(profileStrValues: Collection<String?>) : this(profileStrValues.map {
        it.toString().toLowerCase()
    }.toSet())

    constructor() : this(emptyList())

    val isDev: Boolean
        get() = contains("dev") || contains("development")

    val isProd: Boolean
        get() = contains("prod") || contains("production")

    val isTest: Boolean
        get() = contains("test")
}